src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/ContractNameServices.cs

```diff
@@ -126,7 +130,10 @@ private static void WriteType(StringBuilder typeName, Type type, bool formatGene
                 //
                 Queue<Type> genericTypeArguments = new Queue<Type>(type.GetGenericArguments());
                 WriteGenericType(typeName, type, type.IsGenericTypeDefinition, genericTypeArguments, formatGenericName);
-                Assumes.IsTrue(genericTypeArguments.Count == 0, "Expecting genericTypeArguments queue to be empty.");
+                if(genericTypeArguments.Count != 0) 
+                {
+                    throw new Exception("Expecting genericTypeArguments queue to be empty.");
```

Comment:
Please put this messages and others into the resx file. We can't localize the messages in this state.